### ISSUE TITLE
image_picker how to chose front camera in pickVideo/pickImage

### ISSUE THREAD

**Issue by mnazim** [flutter/flutter#24880]
How do I activate front camera (instead on rear camera) when using `ImagePicker.pickVideo` or `ImagePicker.pickImage`. Front camera can be selected only after picker camera UI is launched.

**Comment by zoechi** [comment]
https://pub.dartlang.org/packages/camera would allow that.

**Comment by agreensh** [comment]
I also this feature would be very useful
e.g. 
imagePicker.hasCameraPosition(ImagePickerCameraPosition.rear); // or ImagePickerCameraPosition.front

imagePicker.setCameraPosition(ImagePickerCameraPosition.rear); // or ImagePickerCameraPosition.front

var cameraPos = imagePicker.getCameraPosition();

**Comment by cyanglaz** [comment]
@mnazim 
Unfortunately, there is no official supported way to do it with the image picker intent (Which we use in our plugin) on Android as far as I know. There are some hacks to do it by sending some extra info to the intent, and they are not supported officially. So we don't plan to support it now, until Android has an official way to do it.

The official way to do it is through the Android camera API mentioned here (https://developer.android.com/training/camera/photobasics)

Since the image_picker is not using the camera API, I don't think we can support that on Android.
If do you need this feature, maybe try to use the [camera](https://pub.dartlang.org/packages?q=camera) to build your custom UI.

If anyone has any suggestions on doing it with image picker intent using an officially supported way, please leave a message in this issue or open another issue to explain. I will be more than happy to implement it.

For now I am closing the issue.

**Comment by 123cristen** [comment]
Even though it's not supported officially, the [existing ways on Android](https://stackoverflow.com/questions/43841738/how-to-open-front-and-back-camera-through-intent-in-android) to switch to the front camera work on most phones I've tested on. This could be a best effort feature anyway, as the user can always choose to flip the camera once it is opened. Happy to submit a pull request for this if it can be supported, thanks!

**Comment by mehmetf** [comment]
@cyanglaz customer: money has asked for this. Is there a way to do this on iOS as well? If so, I vote for supporting it and documenting the Dart API that it is best effort on Android.

**Comment by 123cristen** [comment]
On iOS, I believe it can be supported by specifying [camera device](https://developer.apple.com/documentation/uikit/uiimagepickercontroller/1619117-cameradevice).

**Comment by cyanglaz** [comment]
@mehmetf It is supported on iOS. Reopening the issue. 
I will move this up in my priority list.

**Comment by cyanglaz** [comment]
I created a PR to add this feature here: https://github.com/flutter/plugins/pull/2621
anyone wants to review it?

**Comment by cyanglaz** [comment]
Landed https://github.com/flutter/plugins/pull/2621. We should be able to select the camera if supported by device, starting from `0.6.4` :) im closing the issue.
